src/vivarium/framework/results/manager.py

```diff
@@ -81,6 +83,31 @@ def on_time_step_cleanup(self, event: Event):
     def on_collect_metrics(self, event: Event):
         self.gather_results("collect_metrics", event)
 
+    def on_post_setup(self, event: Event):
```

Comment:
I'd move this to before `def on_time_step_prepare()` (also move the post_setup listener registration to before the the "time_step__prepare" registration.